packages/mongodb-chatbot-ui/src/referenceType.tsx

```diff
@@ -0,0 +1,66 @@
+import { Reference } from "mongodb-rag-core";
+import { MessageDataReference } from "./services/conversations";
+import { RichLinkVariantName } from "@lg-chat/rich-links";
+
+export type ReferenceVariant = RichLinkVariantName;
+
+/**
+  Analyzes a reference link to determine which link variant type to use.
+  @param reference The reference link to analyze.
+  @returns The link variant type to use, or undefined if no variant matches.
+ */
+export function referenceType(ref: Reference): ReferenceVariant | undefined {
```

Comment:
this seems like a reasonable implementation for now, but we should consider having `contentType` as a first-class part of the data model so that you don't need to parse it out from the UI here. 

would be pretty straight forward to include during data ingestion, and serve from server.

...good intern/new hire project. creating ticket - https://jira.mongodb.org/browse/EAI-378